packages/astro/scripts/shiki-gen-languages.mjs

```diff
@@ -4,31 +4,34 @@ const dir = await fs.promises.readdir('packages/astro/node_modules/shiki/languag
 
 const langImports = dir.map((f) => {
 	const key = f.slice(0, f.indexOf('.tmLanguage.json'));
-	return [
-		key,
-		`import('shiki/languages/${f}').then(mod => mod.default).then(grammar => {
-	const lang = BUNDLED_LANGUAGES.find(l => l.id === '${key}');
-	if(lang) {
+	return [key, `import('shiki/languages/${f}').then((mod) => handleLang(mod.default, '${key}'))`];
+});
+
+let code = `\
+/**
+ * This file is prebuilt from packages/astro/scripts/shiki-gen-languages.mjs
+ * Do not edit this directly, but instead edit that file and rerun it to generate this file.
+ */
+
+import { BUNDLED_LANGUAGES } from 'shiki';
+
+function handleLang(grammar, language) {
+	const lang = BUNDLED_LANGUAGES.find((l) => l.id === language);
+	if (lang) {
 		return {
 			...lang,
-			grammar
+			grammar,
 		};
 	} else {
 		return undefined;
 	}
-})`,
-	];
-});
-let code = `import { BUNDLED_LANGUAGES } from 'shiki';
+}
 
+// prettier-ignore
 export const languages = {`;
```

Comment:
Used `// prettier-ignore` here as the codegen code is more readable without prettier's line breaks.